test/Pool.t.sol

```diff
@@ -0,0 +1,668 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import "test/mocks/Token1.sol";
+import "src/Pool.sol";
+import "src/TheToken.sol";
+
+import "forge-std/Test.sol";
+import { console } from "forge-std/console.sol";
+
+contract PoolTest is
+    Test // TO-DO Fixture
+{
+    Pool public pool;
+    Token1 public token1;
+
+    address public deployer = vm.addr(1500);
+    address public user1 = vm.addr(1501);
+    address public user2 = vm.addr(1502);
+    address public user3 = vm.addr(1503);
+    address public user4 = vm.addr(1504);
+
+    function setUp() public {
+        // Label addresses
+        vm.label(deployer, "Deployer");
+        vm.label(user1, "User 1");
+        vm.label(user2, "User 2");
+        vm.label(user3, "User 3");
+        vm.label(user4, "User 4");
+
+        vm.startPrank(deployer);
+
+        token1 = new Token1();
+
+        // Set the initial data
+        // RewardsMultiplier
+        uint64[] memory rewardsMultiplierBlocks = new uint64[](4);
+        rewardsMultiplierBlocks[0] = 100;
+        rewardsMultiplierBlocks[1] = 200;
+        rewardsMultiplierBlocks[2] = 300;
+        rewardsMultiplierBlocks[3] = 400;
+        uint64[] memory rewardsMultipliers = new uint64[](4);
+        rewardsMultipliers[0] = 100;
+        rewardsMultipliers[1] = 50;
+        rewardsMultipliers[2] = 25;
+        rewardsMultipliers[3] = 10;
+
+        // WithdrawFees
+        uint64[] memory withdrawFeeBlocks = new uint64[](4);
+        withdrawFeeBlocks[0] = 0;
+        withdrawFeeBlocks[1] = 100;
+        withdrawFeeBlocks[2] = 1000;
+        withdrawFeeBlocks[3] = 10_000;
+        uint64[] memory withdrawFees = new uint64[](4);
+        withdrawFees[0] = 15;
+        withdrawFees[1] = 10;
+        withdrawFees[2] = 5;
+        withdrawFees[3] = 2;
+
+        pool =
+        new Pool(address(token1), deployer, 20, 10, rewardsMultiplierBlocks, rewardsMultipliers, withdrawFeeBlocks, withdrawFees);
+
+        // Deal the accounts
+        token1.mint(user1, 1000);
+        token1.mint(user2, 1000);
+        token1.mint(user3, 1000);
+        token1.mint(user4, 1000);
+
+        vm.stopPrank();
+    }
+
+    // TO-DO Separate from Pool Logic
+
+    function test_updateBaseRateTokensPerBlock() public {
+        // Unhappy path Nº1 - Trying to update the variable without being the Owner.
+        vm.startPrank(user1);
+
+        vm.expectRevert("Ownable: caller is not the owner");
```

Comment:
nitpick: we usually define a single path for each test but this also looks good tbh